src/Evolution/DiscontinuousGalerkin/Initialization/Mortars.cpp

```diff
@@ -0,0 +1,100 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/DiscontinuousGalerkin/Initialization/Mortars.hpp"
+
+#include <array>
+#include <cstddef>
+#include <tuple>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "Domain/Structure/CreateInitialMesh.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/Element.hpp"
+#include "Evolution/DiscontinuousGalerkin/MortarData.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/MortarHelpers.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Projection.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace evolution::dg::Initialization {
+template <size_t Dim, bool AddFluxBoundaryConditionMortars>
+auto Mortars<Dim, AddFluxBoundaryConditionMortars>::apply_impl(
+    const std::vector<std::array<size_t, Dim>>& initial_extents,
+    const Element<Dim>& element, const TimeStepId& next_temporal_id,
+    const std::unordered_map<Direction<Dim>, Mesh<Dim - 1>>& interface_meshes,
+    const std::unordered_map<Direction<Dim>, Mesh<Dim - 1>>&
+        boundary_meshes) noexcept
+    -> std::tuple<MortarMap<evolution::dg::MortarData<Dim>>,
+                  MortarMap<Mesh<Dim - 1>>,
+                  MortarMap<std::array<Spectral::MortarSize, Dim - 1>>,
+                  MortarMap<TimeStepId>> {
+  static constexpr size_t dim = Dim;
+
+  MortarMap<evolution::dg::MortarData<Dim>> mortar_data{};
+  MortarMap<Mesh<Dim - 1>> mortar_meshes{};
+  MortarMap<std::array<Spectral::MortarSize, Dim - 1>> mortar_sizes{};
+  for (const auto& [direction, neighbors] : element.neighbors()) {
+    for (const auto& neighbor : neighbors) {
+      const auto mortar_id = std::make_pair(direction, neighbor);
+      mortar_data[mortar_id];  // Default initialize data
+      mortar_meshes.emplace(
+          mortar_id, ::dg::mortar_mesh(
+                         interface_meshes.at(direction),
+                         ::domain::Initialization::create_initial_mesh(
+                             initial_extents, neighbor, neighbors.orientation())
+                             .slice_away(direction.dimension())));
+      mortar_sizes.emplace(
+          mortar_id,
+          ::dg::mortar_size(element.id(), neighbor, direction.dimension(),
+                            neighbors.orientation()));
+    }
+  }
+
+  // We will need to update the AddFluxBoundaryConditionMortars to check the
+  // actual boundary condition being imposed.
```

Comment:
This comment isn't very clear to me, because I'm confused about the concept of "update the AddFluxBoundaryConditionMortars".

I'm guessing you're talking about code changes in future PRs where the BCs will be implemented? Perhaps rephrasing to something like "In a future update, we will update the logic below to also check the actual boundary condition being imposed" would be more helpful... to me at least.